tests/language-feature/generics/variadic-partial-inference.slang

```diff
@@ -0,0 +1,25 @@
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK): -shaderobj -output-using-type
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK): -cpu -shaderobj -output-using-type
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK): -vk -shaderobj -output-using-type
+
+//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):out,name=outputBuffer
+RWStructuredBuffer<float> outputBuffer;
+
+void aggregate(inout int v, int u)
+{
+    v = v * u;
+}
+
+T test<T:IInteger, each U : IInteger>(T t, expand each U u)
+{
+    int v = 1;
+    expand aggregate(v, (each u).toInt());
```

Comment:
This is very interesting way of using expend. I guess I didn't understood well enough about how expand was designed.